stl/inc/chrono

```diff
@@ -619,6 +655,1306 @@ namespace chrono {
+    constexpr month_day_last operator/(last_spec, int _Month) noexcept;
+    constexpr month_weekday operator/(const month& _Month, const weekday_indexed& _Wdi) noexcept;
+    constexpr month_weekday operator/(int _Month, const weekday_indexed& _Wdi) noexcept;
+    constexpr month_weekday operator/(const weekday_indexed& _Wdi, const month& _Month) noexcept;
+    constexpr month_weekday operator/(const weekday_indexed& _Wdi, int _Month) noexcept;
+    constexpr month_weekday_last operator/(const month& _Month, const weekday_last& _Wdl) noexcept;
+    constexpr month_weekday_last operator/(int _Month, const weekday_last& _Wdl) noexcept;
+    constexpr month_weekday_last operator/(const weekday_last& _Wdl, const month& _Month) noexcept;
+    constexpr month_weekday_last operator/(const weekday_last& _Wdl, int _Month) noexcept;
+    constexpr year_month_day operator/(const year_month& _Ym, const day& _Day) noexcept;
+    constexpr year_month_day operator/(const year_month& _Ym, int _Day) noexcept;
+    constexpr year_month_day operator/(const year& _Year, const month_day& _Md) noexcept;
+    constexpr year_month_day operator/(int _Year, const month_day& _Md) noexcept;
+    constexpr year_month_day operator/(const month_day& _Md, const year& _Year) noexcept;
+    constexpr year_month_day operator/(const month_day& _Md, int _Year) noexcept;
+    constexpr year_month_day_last operator/(const year_month& _Ym, last_spec) noexcept;
+    constexpr year_month_day_last operator/(const year& _Year, const month_day_last& _Mdl) noexcept;
+    constexpr year_month_day_last operator/(int _Year, const month_day_last& _Mdl) noexcept;
+    constexpr year_month_day_last operator/(const month_day_last& _Mdl, const year& _Year) noexcept;
+    constexpr year_month_day_last operator/(const month_day_last& _Mdl, int _Year) noexcept;
+    constexpr year_month_weekday operator/(const year_month& _Ym, const weekday_indexed& _Wdi) noexcept;
+    constexpr year_month_weekday operator/(const year& _Year, const month_weekday& _Mwd) noexcept;
+    constexpr year_month_weekday operator/(int _Year, const month_weekday& _Mwd) noexcept;
+    constexpr year_month_weekday operator/(const month_weekday& _Mwd, const year& _Year) noexcept;
+    constexpr year_month_weekday operator/(const month_weekday& _Mwd, int _Year) noexcept;
+    constexpr year_month_weekday_last operator/(const year_month& ym, const weekday_last& _Wdl) noexcept;
+    constexpr year_month_weekday_last operator/(const year& _Year, const month_weekday_last& mwdl) noexcept;
+    constexpr year_month_weekday_last operator/(int _Year, const month_weekday_last& mwdl) noexcept;
+    constexpr year_month_weekday_last operator/(const month_weekday_last& mwdl, const year& _Year) noexcept;
+    constexpr year_month_weekday_last operator/(const month_weekday_last& mwdl, int _Year) noexcept;
+
+    // day
+    constexpr day::day(unsigned _Val) noexcept : _Day{static_cast<unsigned char>(_Val)} {}

```

Comment:
I think it would be better if these all functions were defined in place of declaration. Most of them are really small and it would really make this code more readable. Also it'd be more consistent because neither `duration` nor `time_point` move function definitions out of class definition.

Reply:
That's how I originally had it but it started it get tricky to define the `+=` and `-=` operators. Far easier to define them in terms of the non-member `+` and `-` operators. So I moved everything out for consistency.

Reply:
That is actually one of the reasons why I asked this question https://github.com/microsoft/STL/issues/12#issuecomment-555925975 though. 
Because the standard says
```cpp
*this = *this + x
```
and that's not how I was originally defining those.

Reply:
Only observable effects need to be the same. You can just define them as ex. `_Day += _Days.count()` and it would be easier considering that you don't have to keep them outside of class definition.

Reply:
I know, as STL mentioned. As I said, that's how I originally defined them. 

year_month and month_weekday would require a helper function or code duplication unless I'm mistaken? At the time it felt easier to just move what I had.

Reply:
So I remembered what happened. I got to `year_month_day` which kinda needed `year_month_day_last` and that's why I decided to move everything,

So here's what it is with most things declared in the class and together with the declaration.

Couple things that are not where they are declared is `weekday_index/last` returning `weekday::operator[]` as well as the `year_month_day_last` constructor for `year_month_day`.